arch/riscv/adler32_rvv.c

```diff
@@ -12,23 +12,25 @@
 #include "../../zbuild.h"
 #include "../../adler32_p.h"
 
-Z_INTERNAL uint32_t adler32_rvv(uint32_t adler, const uint8_t *buf, size_t len) {
+static inline uint32_t adler32_rvv_impl(uint32_t adler, uint8_t* restrict dst, const uint8_t *src, size_t len, int COPY) {
```

Comment:
These things do end up dispatching to separate function table entries, you can remove the branch here at every loop iteration, even if it is very predictable.

Reply:
This is an inline function shared by `adler32_fold_copy_rvv` and `adler32_rvv` at the bottom of the file.  Each of those gets a private copy of this function with the constant value of `COPY` optimised away.

Reply:
Maybe qualifying that argument as const is maybe a stronger assertion that the branch never happens?  The compiler can see the entire scope of copy but it couldn't hurt.  Somewhat polices the function itself, too.